widget.subscribe.php

```diff
@@ -548,8 +484,61 @@ function add_subscriber() {
 
 	}
 
+	private function extract_subscribe_to_list(): array
+	{
+		if ( isset($_POST['sailthru_email_list']) ) {
+			$sailthru_email_list = sanitize_text_field( $_POST['sailthru_email_list'] );
+		}
+
+		if ( empty($sailthru_email_list) ) {
+			return array('Sailthru Subscribe Widget' => 1); // subscriber is an orphan
+		}
+
+		return $this->create_subscribe_to_list( $sailthru_email_list );
+	}
+
+	private function create_subscribe_to_list( $sailthru_email_list ): array {
```

Comment:
I considered that at first but I kept them separate because my idea here is to keep the first function just extracting the values from post and offloading anything more complex to other functions. The second function then have the responsibility of assembling the List -> subscription_status structure. That said, the name of the second function is bad and I accept suggestions for something more understandable that won't get confused with extract_subscribe_to_list (create_email_list_array?)